container/lxd/container.go

```diff
@@ -262,11 +264,40 @@ func (s *Server) CreateContainerFromSpec(spec ContainerSpec) (*Container, error)
 			Profiles:     spec.Profiles,
 			Devices:      spec.Devices,
 			Config:       spec.Config,
-			Ephemeral:    false,
+			Ephemeral:    ephemeral,
 		},
 	}
 	op, err := s.CreateContainerFromImage(spec.Image.LXDServer, *spec.Image.Image, req)
 	if err != nil {
+		if IsLXDAlreadyExists(err) {
+			container, _, conErr := s.GetContainer(spec.Name)
+			if conErr != nil {
+				return nil, errors.Trace(conErr)
+			}
+
+			// If we don't match the spec from the container, then we're not
+			// sure what we've got here. Return the original error message.
+			if container.Architecture != spec.Architecture ||
+				container.Ephemeral != ephemeral ||
+				!reflect.DeepEqual(container.Profiles, spec.Profiles) ||
+				!reflect.DeepEqual(container.Devices, spec.Devices) ||
+				!reflect.DeepEqual(container.Config, spec.Config) {
+				return nil, errors.Trace(err)
+			}
+
+			switch container.StatusCode {
+			case api.Running:
+				c := Container{*container}
+				return &c, nil
+			case api.Started, api.Starting, api.Success:
+				// If the container isn't running, then send an error to try
+				// and do a retry. Unfortunately, there is no way to wait for
+				// the status of on a running containers.
+				return nil, errors.Errorf("waiting for container %s to be running, got: %d", req.Name, container.StatusCode)
```

Comment:
We should find out how this works in a multi-region lxd scenario.  Does the provisioner do what we think.